extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -1232,6 +1233,7 @@ export class AmpA4A extends AMP.BaseElement {
 
   /** @override */
   layoutCallback() {
+    observe(this.element, (inViewport) => this.viewportCallback_(inViewport));
```

Comment:
For ads it might be important to have this callback prior to `layoutCallback` has fired. Can we confirm this directly? IMHO, it might be worth pushing ads into a separate PR due to a higher probability of rollback and/or more research needed.